pkg/upload/unity-android.go

```diff
@@ -108,6 +122,28 @@ func ProcessUnityAndroid(globalOptions options.CLI, endpoint string, logger log.
 				if filepath.Base(file) == "libil2cpp.sym.so" && utils.ContainsString(fileList, "libil2cpp.dbg.so") {
 					continue
 				}
+
+				// If we're uploading a libil2cpp.so file, we want to extract the build ID so that we can upload the linemapping file
+				if filepath.Base(file) == "libil2cpp.so" && !unityOptions.Shared.NoUploadIl2cppMappingFile {
```

Comment:
The ED says we'll upload this after the .so files, but that if we can't get a build ID then we'll error out.

Suggest splitting this logic to get the path and build ID here (or error out), then upload the .so files, then upload the mapping file at the end.

The ED also mentions using `--verbose` to get more details – I'm not sure if there is any other info we can helpfully provide thought?